### PR TITLE
#396 add visual indication of wildcard

### LINKED ISSUE #396: Add visual indication of ** wildcard?
## Expected Behaviour

In a [support ticket](https://support.msupply.org.nz/cerb6/profiles/ticket/TH-5195-TJ) Lauren (CHAI, Laos) asks whether you can search for an item that contains a phrase. 
image

Ideally we could later add a toggle or some way of changing this in the UI, but until then, do you think it could be useful to have some helper text indicating how it works?

## Current Behaviour

No indication of how to search for "includes x" rather than "starts with x"

### PR DESCRIPTION
**Pull request by katherine-sussol**
Fixes #396

## Description

Add some placeholder and helper text to the explorer UI
Went with @mark-prins  suggestion here: https://github.com/openmsupply/unified-codes/issues/396#issuecomment-846656707
image

I got a bit confused with where to put the props and helper text (also the placeholder text for a code search seems a bit redundant but the filter type is there so...just in case someone tries to use it I guess?)

### Documentation:

Tick below if one of the following applies:

1. Documentation has been added to cover changes introduced by this PR.
2. This PR requires no changes to documentation.

- [x] Documentation OK.

### Tests:

Tick below if one of the following applies:

1. Tests have been added to cover changes introduced by this PR.
2. This PR requires no new tests.

- [x] Tests OK.

### CODE REVIEW

File: unified-codes/apps/web/src/components/containers/explorer/ExplorerSearchBar.tsx [hunk 1]
```diff
@@ -2,17 +2,45 @@ import * as React from 'react';
...
 import { ITheme } from '../../../styles';
 
-const styles = (_: ITheme) => ({
-  input: { paddingLeft: 15 },
-  button: { marginTop: 15 },
-});
+const useStyles = makeStyles((theme: ITheme) =>
+  createStyles({
+    root: { color: theme.palette.text.primary },
+    input: { paddingLeft: 15 },
```
Comment (mark-prins) [inline unified-codes/apps/web/src/components/containers/explorer/ExplorerSearchBar.tsx:15]:
```suggestion
    input: { paddingLeft: 15, color: '#a2a2a2' },
```

I find the white text strange

Screen Shot 2021-05-25 at 9 51 50 AM

and would suggest adding the colour here, and then updating `ClearInput` to use the style. That results in:

Screen Shot 2021-05-25 at 10 18 13 AM

and

Screen Shot 2021-05-25 at 10 18 20 AM

File: unified-codes/apps/web/src/components/containers/explorer/ExplorerPage.tsx [hunk 2]
```diff
@@ -66,7 +66,7 @@ export const ExplorerPageComponent: ExplorerPage = ({
       classes={classes}
       table={<ExplorerTable />}
       toggleBar={<ExplorerToggleBar />}
-      searchBar={<ExplorerSearchBar />}
+      searchBar={<ExplorerSearchBar helperText={'Use * to indicate any characters'} />}
```
Comment (mark-prins) [inline unified-codes/apps/web/src/components/containers/explorer/ExplorerPage.tsx:69]: As this is text, there's no need for `{'`
( I prefer the standard of double-quoting component parameters, though I see that the line below isn't, so you may prefer to go with consistency )

```suggestion
      searchBar={<ExplorerSearchBar helperText="Use * to indicate any characters"/>}
```

File: unified-codes/apps/web/src/selectors/ExplorerSelectors.ts [hunk 3]
```diff
@@ -45,6 +45,17 @@ const selectLabel = createSelector(selectFilterBy, (filterBy: EEntityField): str
   }
 });
 
+const selectPlaceholder = createSelector(selectFilterBy, (filterBy: EEntityField): string => {
+  switch (filterBy) {
+    case EEntityField.CODE:
+      return 'e.g. 4231441c';
+    case EEntityField.DESCRIPTION:
+      return 'e.g. amoxicillin, amox, *mox';
```
Comment (mark-prins) [inline unified-codes/apps/web/src/selectors/ExplorerSelectors.ts:53]: what would be good is to explain that we default to 'starts with' - but I think there is enough instructional text already! I'd let people figure that out for themselves..

File: unified-codes/apps/web/src/components/containers/explorer/ExplorerPage.tsx (hunk 2 above)
Comment (katherine-sussol) [reply unified-codes/apps/web/src/components/containers/explorer/ExplorerPage.tsx]: Oh good point :) thanks

File: unified-codes/apps/web/src/components/containers/explorer/ExplorerSearchBar.tsx (hunk 1 above)
Comment (katherine-sussol) [reply unified-codes/apps/web/src/components/containers/explorer/ExplorerSearchBar.tsx]: Thanks, the fading of the label was a bit odd, this is much nicer :) ee895df5a

File: unified-codes/apps/web/src/selectors/ExplorerSelectors.ts (hunk 3 above)
Comment (katherine-sussol) [reply unified-codes/apps/web/src/selectors/ExplorerSelectors.ts:53]: Hmm. Yeah, I'm conflicted now. It _is_ a bit weird that we default to starts with although it also matches mSupply's weird behaviour...

**Comment by katherine-sussol** [comment]
> Still not sure of the need to limit the classes here.
> I would use
> 
> ```
>       classes={classes}
>       InputLabelProps={{classes}}
> ```
> 
> for these. Either way works though

Thanks, went ahead and changed, easier on the eyes :).

I also edited the helper text a bit after some further thought (although ideally we'd also fix https://github.com/openmsupply/unified-codes/issues/338 and improve the search to use a toggle/checkbox/radio button/something else).

**Pull request merged**